model/godleyIcon.h

```diff
@@ -33,15 +33,70 @@
 namespace minsky
 {
   class GodleyTableEditor;
+  
+  // Godley variables broken out in a separate structure, as copying is non-default
+  class GodleyVars: public ItemT<GodleyVars>
+  {
+  public:	    
+    typedef std::vector<VariablePtr> Variables;
+    const Variables& flowVars() const {return m_flowVars;}
+    const Variables& stockVars() const {return m_stockVars;}    	  
+	  
+    GodleyVars() {}
+    virtual ~GodleyVars() {}
+    // copy operations not deleted to allow ItemT<Group> to compile
+    GodleyVars(const GodleyVars& x) {};
+    GodleyVars& operator=(const GodleyVars&) {return *this;}
+    classdesc::Exclude<std::weak_ptr<GodleyIcon>> self; ///< weak ref to this    
+    
+    void clear() {
+      m_flowVars.clear();
+      m_stockVars.clear();
+    }  
+    
+    bool variableDisplay=true;
+    void toggleVariableDisplay() {variableDisplay=!variableDisplay;}   
+    
+    void removeControlledItems() const override;    
+    
+    /// flows, along with multipliers, appearing in \a col
+    std::map<string,double> flowSignature(int col) const;    
+    
+    GodleyTable table;
+    
+    /// updates the variable lists with the Godley table
+    void update(); 
+    
+    /// returns valueid for variable reference in table
+    // TODO: this should be refactored to a more central location
+    std::string valueId(const std::string& x) const {
+      return VariableValue::valueId(group.lock(), x);
+    }
+    
+    void setCurrency(const std::string& currency) 
+    {for (auto& i: m_stockVars) i->setUnits(currency);}
+    
+    /// performs dimensional analysis on stock var column \a stockName
+    /// @param check indicates whether a consistency check is applied
+    Units stockVarUnits(const std::string& stockName, bool check) const;
+      
+    void insertControlled(Selection& selection) override;    
+    
+  private:
+    friend class SchemaHelper;  
+    void updateVars(Variables& vars, 
+                    const vector<string>& varNames, 
+                    VariableBase::Type varType);
+    /// move contained variables to correct locations within icon
+    void positionVariables() const;
+    Variables m_flowVars, m_stockVars;  
+    CLASSDESC_ACCESS(GodleyVars);
+  };	  
 
-  class GodleyIcon: public ItemT<GodleyIcon>
+  class GodleyIcon: public GodleyVars
   {
-    /// for placement of bank icon within complex
-    float flowMargin=0, stockMargin=0, iconWidth=100, iconHeight=100;
-    /// icon scale is adjusted when Godley icon is resized
-    float m_iconScale=1;
+    friend class GodleyVars;  
```

Comment:
Why is this needed? Any state that needs to be shared should be declared in a protected section of GodleyVars.